### PR TITLE
Arrays are not permitted @QueryParams in JAX-RS.

### PR DESCRIPTION
**Pull request by jsyrjala**
CXF allows them, but Jersey doesn't. Replaced all array @QueryParams
with Lists.

### CODE REVIEW

File: nflow-engine/src/main/java/com/nitorcreations/nflow/engine/workflow/instance/QueryWorkflowInstances.java [hunk 1]
```diff
@@ -138,8 +138,8 @@ public Builder(QueryWorkflowInstances copy) {
      * @param newIds The identifiers.
      * @return this.
      */
-    public Builder addIds(Integer ... newIds) {
-      this.ids.addAll(asList(newIds));
+    public Builder addIds(List<Integer> newIds) {
```
Comment (efonsell) [inline nflow-engine/src/main/java/com/nitorcreations/nflow/engine/workflow/instance/QueryWorkflowInstances.java]: Why Integer... is changed to List<Integer> here?

File: nflow-rest-api/src/main/java/com/nitorcreations/nflow/rest/v1/WorkflowInstanceResource.java [hunk 2]
```diff
@@ -20,9 +20,7 @@
 import io.swagger.annotations.ApiResponses;
 
 import java.net.URI;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
+import java.util.*;
```
Comment (efonsell) [inline nflow-rest-api/src/main/java/com/nitorcreations/nflow/rest/v1/WorkflowInstanceResource.java]: expand this

File: nflow-rest-api/src/main/java/com/nitorcreations/nflow/rest/v1/WorkflowInstanceResource.java [hunk 3]
```diff
@@ -123,8 +121,8 @@ public Response updateWorkflowInstance(@ApiParam("Internal id for workflow insta
   public ListWorkflowInstanceResponse fetchWorkflowInstance(
       @ApiParam("Internal id for workflow instance") @PathParam("id") int id,
       @QueryParam("include") @ApiParam(value = INCLUDE_PARAM_DESC, allowableValues = INCLUDE_PARAM_VALUES, allowMultiple = true) String include) {
-    Collection<ListWorkflowInstanceResponse> instances = listWorkflowInstances(new Integer[] { id }, new String[0], null, null,
-        new String[0], new WorkflowInstanceStatus[0], null, null, include, 1L);
+    Collection<ListWorkflowInstanceResponse> instances = listWorkflowInstances(Arrays.asList(id), Collections.<String>emptyList(), null, null,
```
Comment (efonsell) [inline nflow-rest-api/src/main/java/com/nitorcreations/nflow/rest/v1/WorkflowInstanceResource.java]: static import for Arrays.asList

File: nflow-engine/src/main/java/com/nitorcreations/nflow/engine/workflow/instance/QueryWorkflowInstances.java (hunk 1 above)
Comment (jsyrjala) [inline nflow-engine/src/main/java/com/nitorcreations/nflow/engine/workflow/instance/QueryWorkflowInstances.java]: Other option would be to have something like this in XXXResources

       new QueryWorkflowInstances.Builder().addIds(ids.toArray(new String[ids.length]{}))

Comment (jsyrjala) [inline nflow-engine/src/main/java/com/nitorcreations/nflow/engine/workflow/instance/QueryWorkflowInstances.java]: I guess it is better to have good api in builder and ugly inside Resources.

File: nflow-rest-api/src/main/java/com/nitorcreations/nflow/rest/v1/WorkflowInstanceResource.java (hunk 3 above)
Comment (jsyrjala) [inline nflow-rest-api/src/main/java/com/nitorcreations/nflow/rest/v1/WorkflowInstanceResource.java]: done

File: nflow-rest-api/src/main/java/com/nitorcreations/nflow/rest/v1/WorkflowInstanceResource.java (hunk 2 above)
Comment (jsyrjala) [inline nflow-rest-api/src/main/java/com/nitorcreations/nflow/rest/v1/WorkflowInstanceResource.java]: done

**Pull request merged**